### ISSUE TITLE
Calling new on subclass of BigDecimal returns BigDecimal instance in Ruby 2.6

### ISSUE THREAD

**Issue by jonekdahl** [ruby/bigdecimal#126]
Hi,

I am running some tests with the newly released Ruby 2.6 and there seems to be a difference in behavior when subclassing BigDecimal. I no longer get an instance of the subclass back when calling `<Subclass>.new`, instead I get an instance of BigDecimal:

```
$ rbenv global 2.5.3
$ irb -r bigdecimal
irb(main):001:0> class Foo < BigDecimal; end
=> nil
irb(main):002:0> Foo.new(0).class
=> Foo

$ rbenv global 2.6.0
$ irb -r bigdecimal
irb(main):002:0> class Foo < BigDecimal; end
=> nil
irb(main):003:0> Foo.new(0).class
(irb):3: warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
=> BigDecimal
```

Is this change in behavior intended? Assuming it is, are there other ways to create a subclass of BigDecimal?

I realize that `BigDecimal.new` will be removed, so I need to understand my options here. Perhaps subclassing is not the way to go?

**Comment by mrkn** [comment]
@jonekdahl The easiest way is sticking bigdecimal version to "1.3.5".  I recommend this way if you want to resolve this issue asap.  Please write the following `gem` method call in your script or Gemfile:

```
gem "bigdecimal", "1.3.5"
```

I will provide the way to create instances of the subclass of `BigDecimal` in the version 1.4.3.  Please use it after shipping.

The next version of bigdecimal (it is 2.0.0) does not provide BigDecimal.new, and you cannot create its subclasses.

**Comment by jonekdahl** [comment]
> The next version of bigdecimal (it is 2.0.0) does not provide BigDecimal.new, and you cannot create its subclasses.

Where can I read more about this change?

**Comment by mrkn** [comment]
@jonekdahl I've released version 1.4.3.pre.20190110 including subclassing support.
You can install this new version by the command `gem install bigdecimal --pre`.
Would you examine this version?

**Comment by jonekdahl** [comment]
@mrkn Thank you. I have tested the prerelease version and it works for me.

**Comment by mrkn** [comment]
@jonekdahl Thank you for checking the pre-release version.  Finally, I've released 1.4.3.
